### ISSUE TITLE
overriding spring.main.banner does not work properly

### ISSUE THREAD

**Issue by vvvlc** [spring-projects/spring-boot#17746]
Hi

I have 
```yaml
spring:
    main:
        banner-mode: 'off'
```
in my `application.yml` and when I override `banner-mode` on command line via 
```sh
java -jar .... --spring.main.banner-mode=console
```
Then banner is shown twice, this is an incorrect behavior. 
```
vlcvi01@VLCVI01W10:~/src/apl$ java -jar gateway-service.jar --spring.main.banner-mode=console
2019-07-31 12:26:21.644 <ZWEAGW1:main: > vlcvi01 INFO  (c.c.m.p.s.BuildInfo,BuildInfo.java:23) Service gateway-service version 1.1.7-SNAPSHOT #n/a on null by VLCVI01W10 commit 317669d
           _____ _____    _____       _
     /\   |  __ \_   _|  / ____|     | |
    /  \  | |__) || |   | |  __  __ _| |_ _____      ____ _ _   _
   / /\ \ |  ___/ | |   | | |_ |/ _` | __/ _ \ \ /\ / / _` | | | |
  / ____ \| |    _| |_  | |__| | (_| | ||  __/\ V  V / (_| | |_| |
 /_/    \_\_|   |_____|  \_____|\__,_|\__\___| \_/\_/ \__,_|\__, |
                                                             __/ |
                                                            |___/

           _____ _____    _____       _
     /\   |  __ \_   _|  / ____|     | |
    /  \  | |__) || |   | |  __  __ _| |_ _____      ____ _ _   _
   / /\ \ |  ___/ | |   | | |_ |/ _` | __/ _ \ \ /\ / / _` | | | |
  / ____ \| |    _| |_  | |__| | (_| | ||  __/\ V  V / (_| | |_| |
 /_/    \_\_|   |_____|  \_____|\__,_|\__\___| \_/\_/ \__,_|\__, |
                                                             __/ |
                                                            |___/

2019-07-31 12:26:28.511 <ZWEAGW1:main:28067> vlcvi01 INFO  (o.s.b.w.e.t.TomcatWebServer,Jdk14Logger.java:87) Tomcat initialized with port(s): 10010 (https)
2019-07-31 12:26:28.537 <ZWEAGW1:main:28067> vlcvi01 INFO  (o.a.c.h.Http11NioProtocol,DirectJDKLog.java:180) Initializing ProtocolHandler ["https-jsse-nio-127.0.0.1-10010"]
```
If I use 
```sh
java -jar .... --spring.main.banner-mode=log
```
no banner is shown.
```
vlcvi01@VLCVI01W10:~/src/apl$ vlcvi01@VLCVI01W10:~/src/apl$ java -jar gateway-service.jar --spring.main.banner-mode=log
2019-07-31 12:25:28.009 <ZWEAGW1:main: > vlcvi01 INFO  (c.c.m.p.s.BuildInfo,BuildInfo.java:23) Service gateway-service version 1.1.7-SNAPSHOT #n/a on null by VLCVI01W10 commit 317669d
2019-07-31 12:25:34.961 <ZWEAGW1:main:27988> vlcvi01 INFO  (o.s.b.w.e.t.TomcatWebServer,Jdk14Logger.java:87) Tomcat initialized with port(s): 10010 (https)
```

**Comment by mbhave** [comment]
@vvvlc I'm not able to reproduce this behavior. I set `spring.main.banner-mode: 'off'` in `application.yml`. Running `java -jar ..` without any command line parameters results in no banner being shown. If I run it with `java -jar .... --spring.main.banner-mode=console`, the banner is printed to the console (once not twice).

Please provide a minimal sample that we can run to reproduce the issue.

**Comment by vvvlc** [comment]
I agree with @mbhave  I was not able to recreate it on minimal springboot app like https://github.com/jadekler/git-misc-2/tree/master/boot-banner-properties-off

Our situation is that we would like to disable banner in application.yml and enable it via additional profile if needed, when we reenabled it then we got double banner.

Here is our code, https://github.com/zowe/api-layer/pull/341 but it could be very difficult to extract it into something minimal, more over @ilikinabdullayev provide nice workaround in using default value when a parameters is not defined.
I close case, sorry for wasted your time it is probably caused other netflix components in our app.
Vitek
